### PR TITLE
Add test matrix Docker images for OpenJ9

### PR DESCRIPTION
**Pull request by agoallikmaa**
Added OpenJ9 test image creation for all existing appservers. These images have been tested to pass smoke tests locally, but that will be a separate PR because these images will only exist in the public repository after this is merged.

Changed the way that images are listed to be both more flexible (allows overriding Dockerfile name, and give custom build arguments per version/vm/jdk) and avoid VM name and version number duplication for matrix-like groups (without exclusions).

### CODE REVIEW

File: smoke-tests/matrix/src/payara-custom-5.2020.6.dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    PREBOOT_COMMANDS="${HOME_DIR}/config/pre-boot-commands.asadmin" \
+    POSTBOOT_COMMANDS="${HOME_DIR}/config/post-boot-commands.asadmin" \
+    PATH="${PATH}:${HOME_DIR}/scripts"
+
+COPY --from=default $HOME_DIR $HOME_DIR
+RUN rm ${PAYARA_DIR}/glassfish/modules/phonehome-bootstrap.jar
+
+WORKDIR $HOME_DIR
+
+COPY app.war $DEPLOY_DIR
```
Comment (iNikem) [inline smoke-tests/matrix/src/payara-custom-5.2020.6.dockerfile:31]: Move COPY line to the end of the file. Then static EXPOSE and CMD layers can be cached.

File: smoke-tests/matrix/src/tomee-custom.dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+FROM tomee:${jdk}-jre-${version}-webprofile as default
+
+FROM adoptopenjdk:${jdk}-jdk-${vm}
+
+ENV SERVER_BASE=/usr/local/tomee
+COPY --from=default $SERVER_BASE $SERVER_BASE
+WORKDIR $SERVER_BASE
+
+COPY app.war $SERVER_BASE/webapps/
```
Comment (iNikem) [inline smoke-tests/matrix/src/tomee-custom.dockerfile:13]: Move COPY line to the end of the file. Then static EXPOSE and CMD layers can be cached.

**Review by iNikem — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

File: smoke-tests/matrix/src/payara-custom-5.2020.6.dockerfile (hunk 1 above)
Comment (agoallikmaa) [reply smoke-tests/matrix/src/payara-custom-5.2020.6.dockerfile]: Done.

File: smoke-tests/matrix/src/tomee-custom.dockerfile (hunk 2 above)
Comment (agoallikmaa) [reply smoke-tests/matrix/src/tomee-custom.dockerfile]: Done.

**Pull request merged**